sources/packages/backend/libs/integrations/src/services/sfas/sfas-restriction.service.ts

```diff
@@ -45,10 +50,21 @@ export class SFASRestrictionService
         select: { id: true },
         where: { restrictionCode: "LGCY" },
       });
-      await this.repo.manager.query(this.bulkInsertStudentRestrictionsSQL, [
-        legacyRestriction.id,
-        creator.id,
-      ]);
+      await this.repo.manager.transaction(async (manager) => {
```

Comment:
I believe that most of the time we start transactions like this from the `dataSource`.